confgenerator/testdata/valid/linux/logging-receiver_mongodb/golden_fluent_bit_main.conf

```diff
@@ -0,0 +1,300 @@
+    Condition Key_Value_Equals severity I
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity ERROR
+    Condition Key_Value_Equals severity E
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity FATAL
+    Condition Key_Value_Equals severity F
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity WARNING
+    Condition Key_Value_Equals severity W
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Match  mongo.mongodb_default
+    Name   modify
+    Remove severity
+
+[FILTER]
+    Match  mongo.mongodb_default
+    Name   modify
+    Rename c component
+
+[FILTER]
+    Match  mongo.mongodb_default
+    Name   modify
+    Rename ctx context
+
+[FILTER]
+    Match  mongo.mongodb_default
+    Name   modify
+    Rename msg message
+
+[FILTER]
+    Key_Name message
+    Match    mongo.mongodb_default
+    Name     parser
+    Parser   mongo.mongodb_default_regex.mongodb
+
+[FILTER]
+    Match  mongo.mongodb_default
+    Name   modify
+    Rename s severity
+
+[FILTER]
+    Add       logging.googleapis.com/severity DEBUG
+    Condition Key_Value_Equals severity D
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity DEBUG
+    Condition Key_Value_Equals severity D1
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity DEBUG
+    Condition Key_Value_Equals severity D2
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity DEBUG
+    Condition Key_Value_Equals severity D3
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity DEBUG
+    Condition Key_Value_Equals severity D4
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity DEBUG
+    Condition Key_Value_Equals severity D5
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity INFO
+    Condition Key_Value_Equals severity I
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity ERROR
+    Condition Key_Value_Equals severity E
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity FATAL
+    Condition Key_Value_Equals severity F
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Add       logging.googleapis.com/severity WARNING
+    Condition Key_Value_Equals severity W
+    Match     mongo.mongodb_default
+    Name      modify
+
+[FILTER]
+    Match  mongo.mongodb_default
+    Name   modify
+    Remove severity
+
+[FILTER]
```

Comment:
The filters from line 65 - 275 seem pretty expensive.

Can we use the built-in feature like severity_key instead? https://github.com/fluent/fluent-bit/blob/0ba2dc2196e4c2c19a6075f8154afe7be137119d/plugins/out_stackdriver/stackdriver_conf.c#L293

Reply:
In fact, that might having this app use its own output plugin instance, which comes with some concerns as well.

I'm trying to think what other ways to reduce the complexity, when I noticed that

https://github.com/GoogleCloudPlatform/ops-agent/pull/287/files#diff-56a349cfc933ac75b1c65de7a835cfce442abcb88d21628c7c4db0c240620a61R119-R177 and https://github.com/GoogleCloudPlatform/ops-agent/pull/287/files#diff-56a349cfc933ac75b1c65de7a835cfce442abcb88d21628c7c4db0c240620a61R210-R268 seem identical? Do we need both?

Reply:
As far as the severity key link you posted, it is looking to process the stackdriver output of the fluent bit entry right? As in it is looking for a "severity_key" on the entry or the [default](https://github.com/fluent/fluent-bit/blob/0ba2dc2196e4c2c19a6075f8154afe7be137119d/plugins/out_stackdriver/stackdriver.h#L55).

What these lines are doing is mapping severity to their respective https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#logseverity GCL severity. 

As for the second comment about the similar blocks, they are essentially doing two of the same things although we need both of them because we have two parsers being run because we try to auto-detect between json and regex based logs which in fluent bit seems to take the approach of just add a separate parser for each.

Reply:
You ended up being right these were getting duplicated on both parsers. After re-examining I shared the severity parser portion on both the JSON and Regex based logs 👍

Reply:
Great, thanks!